packages/flutter_test/lib/src/test_pointer.dart

```diff
@@ -437,3 +437,29 @@ class TestGesture {
     });
   }
 }
+
+/// A record of input [PointerEvent] list with the timeStamp of when it is
+/// injected.
+///
+/// The [timeStamp] is used to indicate the time when the pack is received.
+///
+/// This is a simulation of how the framework is receiving input events from
+/// the engine. See [GestureBinding] and [PointerDataPacket].
+class PointerEventRecord {
+  /// Creates a pack of [PointerEvent]s.
+  PointerEventRecord(this.timeStamp, this.events);
+
+  /// The time stamp of when the event record is received.
+  ///
+  /// This value is used as the time delay relative to the start of
+  /// [WidgetTester.handlePointerEventRecord] call.
+  final Duration timeStamp;
```

Comment:
nit: is there a clearer name we could use for this? Maybe just `delay`?

Reply:
The reason I'm calling it a timeStamp is the data comes from timeline timestamp in my recording package. But since there's no recording context here maybe I need another name. I don't like `delay` because it's too similar to `delayed` that's used `Future` everywhere. Maybe `timeDelay`.

Reply:
timeDelay SGTM